packages/common/test/directives/ng_optimized_image_spec.ts

```diff
@@ -662,406 +665,567 @@ describe('Image directive', () => {
+                  `${IMG_BASE_URL}/path/img.png 1080w, ${IMG_BASE_URL}/path/img.png 1200w, ${
+                      IMG_BASE_URL}/path/img.png 1920w, ` +
+                  `${IMG_BASE_URL}/path/img.png 2048w, ${IMG_BASE_URL}/path/img.png 3840w`);
+        });
+      });
+
+      describe('placeholder attribute', () => {
+        const imageLoader = (config: ImageLoaderConfig) => {
+          let paramsString = '';
+          if (config.isPlaceholder) {
+            paramsString = 'ph=true';
+          }
+          let queryString = `${config.width ? 'w=' + config.width + '&' : ''}${paramsString}`;
+          return `${IMG_BASE_URL}/${config.src}?${queryString}`;
+        };
+
+        it('should add background size, position, and repeat styling if the placeholder attribute is present',
+           () => {
+             setupTestingModule();
+             const template =
+                 '<img ngSrc="path/img.png" width="400" height="300" placeholder="path/img.png" />';
+
+             const fixture = createTestComponent(template);
+             fixture.detectChanges();
+             const nativeElement = fixture.nativeElement as HTMLElement;
+             const img = nativeElement.querySelector('img')!;
+             const styles = parseInlineStyles(img);
+             expect(styles.get('background-size')).toBe('cover');
+             expect(styles.get('background-position')).toBe('50%50%');
+             expect(styles.get('background-repeat')).toBe('no-repeat');
+           });
+
+        it('should not add background styles if the placeholder attribute is not present', () => {
+          setupTestingModule();
+          const template = '<img ngSrc="path/img.png" width="400" height="300">';
+
+          const fixture = createTestComponent(template);
+          fixture.detectChanges();
+          const nativeElement = fixture.nativeElement as HTMLElement;
+          const img = nativeElement.querySelector('img')!;
+          const styles = parseInlineStyles(img);
+          expect(styles.get('background-size')).toBeUndefined();
+          expect(styles.get('background-position')).toBeUndefined();
+          expect(styles.get('background-repeat')).toBeUndefined();
+          expect(styles.get('background-image')).toBeUndefined();
+        });
+
+        it('should add a background-image tag when placeholder is provided as a boolean', () => {
+          setupTestingModule({imageLoader});
+          const template = '<img ngSrc="path/img.png" width="400" height="300" placeholder=true />';
+
+          const fixture = createTestComponent(template);
+          fixture.detectChanges();
+          const nativeElement = fixture.nativeElement as HTMLElement;
+          const img = nativeElement.querySelector('img')!;
+          const styles = parseInlineStyles(img);
+          expect(styles.get('background-image'))
+              .toBe('url("http://localhost:9876/path/img.png?w=20&ph=true")');
```

Comment:
I'd propose using `IMG_BASE_URL` to make sure this assert works in different environments.
```suggestion
              .toBe(`url("${IMG_BASE_URL}/path/img.png?w=20&ph=true")`);
```

Reply:
Whoops, fixed in several places in the test